src/__test__/utilities.test.ts

```diff
@@ -0,0 +1,16 @@
+import { allocateSubjectColor } from '../utilities';
+
+it('allocateSubjectColor first 6 colors have no duplicates.', () => {
+  // Repeat 20 times to avoid getting lucky with randomization.
+  for (let i = 0; i < 20; i += 1) {
+    const subjectColor: Record<string, string> = {};
+    allocateSubjectColor(subjectColor, 'CS');
+    allocateSubjectColor(subjectColor, 'MATH');
+    allocateSubjectColor(subjectColor, 'ECE');
+    allocateSubjectColor(subjectColor, 'INFO');
+    allocateSubjectColor(subjectColor, 'ISST');
+    allocateSubjectColor(subjectColor, 'ORIE');
+    expect(new Set(Object.values(subjectColor)).size).toBe(6);
+  }
+  // Cannot test other things since object key order is unreliable.
```

Comment:
See my other comment about `allocateSubjectColor`; why is it important that the first 6 colors have no duplicates if the colors are pre-allocated for all (>>6) subjects?

If we still go with this implementation, I think you may still want to test >6 function calls because it's an edge case. For example, allocateSubjectColor 7 times and ensure that the 7th subject's color is valid / equal to one of the first 6. This can guard against errors in the `unusedSubjectColors` and `subjectColorCandidates` assignments.
I think the function is correct though.

Reply:
I'm mostly trying to preserve the current behavior.